helpers.c

```diff
@@ -26,9 +29,68 @@ int sync_file_range(int fd, uint64_t offset, uint64_t nbytes,
 }
 #endif
 
+#ifdef CONFIG_MAC_FADVISE_DISCARD
+/**
+ * CONFIG_MAC_FADVISE_DISCARD should not be run within Rosetta. Native arm64 or x86 only.
+ */
+
+#define MMAP_CHUNK_SIZE		(1024 * 1024 * 1024)
+
+int discard_pages(int fd, off_t offset, off_t size)
+{
+	caddr_t *addr;
+	uint64_t chunk_size = MMAP_CHUNK_SIZE;
+
+	printf("fsync'ing file, then invalidating UBC.\n");
+	if (fsync(fd) < 0) {
+		fprintf(stderr, "Cannot fsync file\n");
+		exit(1);
```

Comment:
Needs to return an error here, not exit. It should just return errno, but before you overwrite it by calling fprintf.